File: admin/client/tools/duckduckgo.ts
import { z } from "zod";

export const DuckDuckGoToolConfig = z.object({
  name: z.literal("duckduckgo"),
  label: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  enabled: z.boolean().nullable().optional(),
  config: z.object({}).nullable().optional(),
});
export type DuckDuckGoToolConfigType = z.infer<typeof DuckDuckGoToolConfig>;
export const DEFAULT_DUCKDUCKGO_TOOL_CONFIG = {
  label: "DuckDuckGo",
  description: "",
  config: {},
  enabled: false,
};

File: src/ragapp/backend/models/tools/duckduckgo.py
from typing import ClassVar, Dict, Literal, Optional

from pydantic import BaseModel


class DuckDuckGoTool(BaseModel):
    config_id: ClassVar[str] = "duckduckgo"
    name: Literal["duckduckgo"] = "duckduckgo"
    tool_type: Literal["local"] = "local"
    label: Literal["Web Search"] = "Web Search"
    description: str = (
        "Search more information about a topic from the query using DuckDuckGo."
    )
    custom_prompt: Optional[str] = None
    config: Dict = {}
    enabled: bool = False
